### PR TITLE
Optimize markdown parsing

### LINKED ISSUE #4026: tests: the sub items (with 3 hash marks `###`) are staying in the main list
not respecting the hierarchy and making it impossible to test the alphabetical order

https://github.com/avelino/awesome-go/blob/5f680d4c62c7275d40191a05c7772f17a0b2a1ae/repo_test.go#L23-L27

### PR DESCRIPTION
**Pull request by benschoch**
Suggestion to replace the markdown parser with a more modular one – also from the Awesome list of course ;)
This fixes several rendering issues and also allows
fixed: #4026

fix-nested-lists

### CODE REVIEW

**Comment by avelino** [comment]
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body links content check:
- godoc.org or pkg.go.dev: True
- goreportcard.com: False
- coverage: False

> your project is about to be approved, it's under revision, _it may take a few days_

File: make_site.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 	"text/template"
 
 	"github.com/PuerkitoBio/goquery"
-	"github.com/avelino/slugify"
+	"github.com/avelino/awesome-go/pkg/slug"
```
Comment (avelino) [inline make_site.go:13]: this package `github.com/avelino/slugify` not only treat `/` by `""`, it has other special character treatments

**Review by avelino — CHANGES_REQUESTED** [review]
we need to have test coverage for the new packages that have been created

File: make_site.go (hunk 1 above)
Comment (benschoch) [reply make_site.go:13]: Sure, it does. That's why it is still used and just wrapped within `awesome-go/pkg/slug` to add the more GitHub style output. Would you prefer to move it over to the `avelino/slugify` package or what do you mean?

File: make_site.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 	"text/template"
 
 	"github.com/PuerkitoBio/goquery"
-	"github.com/avelino/slugify"
```
Comment (Henrryxdjr) [inline make_site.go:13]: Ayuda

**Review by Henrryxdjr — APPROVED** [review]

**Comment by noahescobido16-cloud** [comment]
Pa follow guys

**Pull request merged**